src/Microsoft.AspNet.Mvc.Core/Infrastructure/MvcRouteHandler.cs

```diff
@@ -68,12 +63,13 @@ public VirtualPathData GetVirtualPath(VirtualPathContext context)
                         context.RouteData.Values.Add(kvp.Key, kvp.Value);
                     }
                 }
-            }
 
-            // Removing RouteGroup from RouteValues to simulate the result of conventional routing
-            context.RouteData.Values.Remove(TreeRouter.RouteGroupKey);
+                // Removing RouteGroup from RouteValues to simulate the result of conventional routing
+                context.RouteData.Values.Remove(TreeRouter.RouteGroupKey);
```

Comment:
Any particular reason this moved to inside the `if()`? Is it not needed in other cases?